lib/aiken/rationals.ak

```diff
@@ -0,0 +1,628 @@
+//// This module implements operations between data structures of type Rational.
+////
+//// Rational numbers are represented as a pair of integers, a numerator and a denominator.
+////
+//// - the denominator is always positive
+//// - the sign is managed in the numerator.
+//// - the denominator is never 0.
+//// - the numerator and denominator are not explicitly reduced. This is done on demand.
+////
+//// Reducing should be done before write (e.g validating datum output)
+
+// todo: more comments (good docs)
+//      finish compare tests
+
+use aiken/math
+use aiken/option
+
+/// Useful for using Rationals in datums/redeemers which cannot have opaque types.
+/// 
+/// Use **check_rational** or **unsafe_check_rational** to convert to a Rational.
+pub type UncheckedRational {
+  numerator: Int,
+  denominator: Int,
+}
```

Comment:
I am still not convinced by this type in the stdlib. It'd leave it to users / projects to define it should they want to. And I'd suggest simply having one constructor function in the stdlib (instead of `check_rational`):

```gleam
pub fn new(numerator: Int, denominator: Int) -> Option<Rational>
```

Reply:
Yea new is good